karoo_gp/base_class.py

```diff
@@ -159,7 +159,26 @@ def __init__(
         self.X_hash = None                   # hash of last-used fit data
         self.gen_max = gen_max               # number of generations to evolve
         self.swim = swim                     # culling method
-        self.tree_depth_max = tree_depth_max # max allowed depth
+
+        if tree_depth_max is None:           # max allowed depth
+            default_max = 4
```

Comment:
I don't like too much the default hardcoded here.

Since now the `tree_depth_max` defaults to `None` both in the `argparse` and in the `__init__` args, maybe we can just assume that the `tree_depth_max` is set to whatever the `tree_depth_base` is unless otherwise specified, and get rid of both the default of `4` and the warning.

* `tree_depth_base` is either the default (`4`) or the value passed by the user
* if `tree_depth_max` is not specified:
  * set `tree_depth_max = tree_depth_base`
* if `tree_depth_max` is specified:
  * if `tree_depth_max < tree_depth_base`: raise an error
  * if `tree_depth_max >= tree_depth_base`: everything is ok